File: LivrusVirtus/FoxToolKit/FXRuby/CadAssuntos.rb
require 'fox'
require 'fox/responder'
require 'Cadastros'
require 'Principal'
require 'RotinasGlobais'
require 'ConsultasSQL'
include Fox
include Responder

class FRMCadAssuntos < FRMCadastros
  
  def initialize(clienteMDI, iconeMDI, menuMDI)  
    super(clienteMDI, Resource::STR_CADASSUNTO, iconeMDI, menuMDI, 
      TEXTFIELD_INTEGER, true)

    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNNOVO, :onBtnNovo)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNPESQUISAR, :onBtnPesquisar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNANTERIOR, :onBtnAnterior)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNPROXIMO, :onBtnProximo)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNSALVAR, :onBtnSalvar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNEXCLUIR, :onBtnExcluir)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNLIMPAR, :onBtnLimpar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNFECHAR, :onBtnFechar)
    FXMAPFUNC(SEL_CLOSE, 0, :onClose)

    @labDescricao.setText("Assunto: ")
  end

  def ValidaDados(bTodosDados)
    if $Rotinas.ValidaCampo(@edtCodigo.getText(), Resource::STR_CODNINF)
    	  return false
    end
  	if bTodosDados
  	  if $Rotinas.ValidaCampo(@edtDescricao.getText(), Resource::STR_ASSNINF)
    		return false
      end
  	end
  	return true
  end 
  
  def onBtnNovo(sender, sel, ptr)
    btnNovoClick()
    NovoDado($ConsSQL.Assunto('N', @edtCodigo.getText(),""))	
    return true
  end
  
  def onBtnPesquisar(sender, sel, ptr)
    PesquisarDados(
      $ConsSQL.Assunto('S', @edtCodigo.getText(),""), Resource::STR_ASSNENC)
    return true
  end
  
  def onBtnAnterior(sender, sel, ptr)
    registroAnterior()	
    return true
  end
  
  def onBtnProximo(sender, sel, ptr)
    registroProximo()
    return true
  end
  
  def onBtnSalvar(sender, sel, ptr)
    SalvarDados(ValidaDados(true),
      $ConsSQL.Assunto('S', @edtCodigo.getText(),""),
      $ConsSQL.Assunto('U', @edtCodigo.getText(), @edtDescricao.getText()),
      $ConsSQL.Assunto('I', @edtCodigo.getText(), @edtDescricao.getText()), 
      true)
    return true
  end
  
  def onBtnExcluir(sender, sel, ptr)
    ExcluirDados(ValidaDados(false),
      $ConsSQL.Assunto('D', @edtCodigo.getText(),""))
    return true
  end
  
  def onBtnLimpar(sender, sel, ptr)
    btnLimparClick()
    return true
  end

  def onBtnFechar(sender, sel, ptr)
    onClose(sender, sel, ptr)
    return true
  end

  def onClose(sender, sel, ptr)
    $frmPrincipal.habilitaMenu("CadAssuntos")
    destroy()
    return true
  end 
 
end

File: LivrusVirtus/FoxToolKit/FXPython/CadAssuntos.py
from FXPy.fox import *
from Cadastros import frmCadastros
from resource import Resource
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
import Principal

class frmCadAssuntos(frmCadastros):

  def __init__(self, clienteMDI, iconeMDI, menuMDI):  
    frmCadastros.__init__(self, clienteMDI, Resource.STR_CADASSUNTO, 
      iconeMDI, menuMDI, TEXTFIELD_INTEGER, True)

    self.Rotinas = RotinasGlobais.Rotinas
    self.ConsSQL = ConsultasSQL.ConsSQL

    FXMAPFUNC(self, SEL_COMMAND, frmCadAssuntos.ID_FRMCADASTROS_BTNNOVO, 
      frmCadAssuntos.onBtnNovo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAssuntos.ID_FRMCADASTROS_BTNPESQUISAR, 
      frmCadAssuntos.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAssuntos.ID_FRMCADASTROS_BTNANTERIOR, 
      frmCadAssuntos.onBtnAnterior)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAssuntos.ID_FRMCADASTROS_BTNPROXIMO, 
      frmCadAssuntos.onBtnProximo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAssuntos.ID_FRMCADASTROS_BTNSALVAR, 
      frmCadAssuntos.onBtnSalvar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAssuntos.ID_FRMCADASTROS_BTNEXCLUIR, 
      frmCadAssuntos.onBtnExcluir)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAssuntos.ID_FRMCADASTROS_BTNLIMPAR, 
      frmCadAssuntos.onBtnLimpar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadAssuntos.ID_FRMCADASTROS_BTNFECHAR, 
      frmCadAssuntos.onBtnFechar)
    FXMAPFUNC(self, SEL_CLOSE, 0, frmCadAssuntos.onClose)

    self.labDescricao.setText("Assunto: ")

  def ValidaDados(self, bTodosDados):
    if self.Rotinas.ValidaCampo(self.edtCodigo.getText(), Resource.STR_CODNINF):
   	  return False
    if bTodosDados:
  	  if self.Rotinas.ValidaCampo(self.edtDescricao.getText(), Resource.STR_ASSNINF):
    		return False        	
    return True
  
  def onBtnNovo(self, sender, sel, ptr):
    self.btnNovoClick()
    self.NovoDado(self.ConsSQL.Assunto('N', self.edtCodigo.getText(),""))	
    return True  
  
  def onBtnPesquisar(self, sender, sel, ptr):
    self.PesquisarDados(
      self.ConsSQL.Assunto('S', self.edtCodigo.getText(),""), Resource.STR_ASSNENC)
    return True  
  
  def onBtnAnterior(self, sender, sel, ptr):
    self.registroAnterior()	
    return True  
  
  def onBtnProximo(self, sender, sel, ptr):
    self.registroProximo()
    return True  
  
  def onBtnSalvar(self, sender, sel, ptr):
    self.SalvarDados(self.ValidaDados(True),
      self.ConsSQL.Assunto('S', self.edtCodigo.getText(),""),
      self.ConsSQL.Assunto('U', self.edtCodigo.getText(), 
        self.edtDescricao.getText()),
      self.ConsSQL.Assunto('I', self.edtCodigo.getText(), 
        self.edtDescricao.getText()), 
      True)
    return True  
  
  def onBtnExcluir(self, sender, sel, ptr):
    self.ExcluirDados(self.ValidaDados(False),
      self.ConsSQL.Assunto('D', self.edtCodigo.getText(),""))
    return True  
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.btnLimparClick()
    return True  

  def onBtnFechar(self, sender, sel, ptr):
    self.onClose(sender, sel, ptr)
    return True
  
  def onClose(self, sender, sel, ptr):
    Principal.fPrincipal.habilitaMenu("CadAssuntos")
    self.destroy()
    return True
